### PR TITLE
UDS through HttpClient (net5.0+)

### PR DESCRIPTION
**Pull request by colin-higgins**
Use built in overrides in net5.0

@DataDog/apm-dotnet

### CODE REVIEW

File: tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs [hunk 1]
```diff
@@ -125,16 +125,6 @@ public static void Initialize()
                 Log.Error(ex, ex.Message);
             }
 
-            try
-            {
-                var asm = typeof(Instrumentation).Assembly;
-                Log.Information($"[Assembly metadata] Location: {asm.Location}, CodeBase: {asm.CodeBase}, GAC: {asm.GlobalAssemblyCache}, HostContext: {asm.HostContext}, SecurityRuleSet: {asm.SecurityRuleSet}");
```
Comment (lucaspimentel) [inline tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs:131]: Was this removed intentionally?

File: tracer/src/Datadog.Trace/FrameworkDescription.NetCore.cs [hunk 2]
```diff
@@ -93,12 +93,19 @@ private static string GetNetCoreOrNetFrameworkVersion()
             {
                 try
                 {
+#if NETFX
```
Comment (lucaspimentel) [inline tracer/src/Datadog.Trace/FrameworkDescription.NetCore.cs:96]: I think the "canonical" property is `NETFRAMEWORK`
```suggestion
#if NETFRAMEWORK
```

Comment (lucaspimentel) [inline tracer/src/Datadog.Trace/FrameworkDescription.NetCore.cs:96]: What is `NETFX`?

Comment (lucaspimentel) [reply tracer/src/Datadog.Trace/FrameworkDescription.NetCore.cs:96]: This entire file is .NET Core only, so I think you can just change `RootAssembly.CodeBase` to `RootAssembly.Location` below, if that was the intent.

File: tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs (hunk 1 above)
Comment (colin-higgins) [reply tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs:131]: Yes, I added this during functions work and I'm not sure it adds value. Also the GAC references fail in net5.0.

File: tracer/src/Datadog.Trace/FrameworkDescription.NetCore.cs (hunk 2 above)
Comment (colin-higgins) [reply tracer/src/Datadog.Trace/FrameworkDescription.NetCore.cs]: Fixed

**Comment by kevingosse** [comment]
I believe the NET5 target and the UDS work should be in two separate PRs for proper review.

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #2399 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa): 
* 1 benchmarks are **slower**, with geometric mean 1.151
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`WriteAndFlushEnrichedTraces`|net472|830μs|2.04μs|7.06μs|0| 0|0|3.16&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|648μs|2.45μs|9.51μs|0| 0|0|2.57&nbsp;KB|
|#2399|`WriteAndFlushEnrichedTraces`|net472|830μs|3.24μs|11.7μs|0| 0|0|3.16&nbsp;KB|
|#2399|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|649μs|2.57μs|9.97μs|0| 0|0|2.57&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Unknown :shrug: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`SendRequest`|netcoreapp3.1|285μs|1.05μs|4.07μs|0.145| 0|0|19.67&nbsp;KB|
|#2399|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|#2399|`SendRequest`|netcoreapp3.1|301μs|1.2μs|4.66μs|0.144| 0|0|19.67&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`ExecuteNonQuery`|net472|1.57μs|7.52ns|29.1ns|0.093| 0.000781|0|586&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`ExecuteNonQuery`|netcoreapp3.1|1.43μs|5.47ns|21.2ns|0.00864| 0|0|624&nbsp;B|
|#2399|`ExecuteNonQuery`|net472|1.58μs|8.06ns|36.9ns|0.0927| 0.000766|0|586&nbsp;B|
|#2399|`ExecuteNonQuery`|netcoreapp3.1|1.46μs|4.96ns|19.2ns|0.0085| 0|0|624&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`CallElasticsearch`|net472|2.39μs|12.3ns|56.3ns|0.124| 0|0|794&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`CallElasticsearch`|netcoreapp3.1|1.57μs|5.42ns|20.3ns|0.0107| 0|0|784&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`CallElasticsearchAsync`|net472|2.51μs|13.8ns|76.7ns|0.146| 0|0|931&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`CallElasticsearchAsync`|netcoreapp3.1|1.77μs|9.33ns|50.2ns|0.0126| 0|0|904&nbsp;B|
|#2399|`CallElasticsearch`|net472|2.28μs|5.49ns|19.8ns|0.124| 0|0|794&nbsp;B|
|#2399|`CallElasticsearch`|netcoreapp3.1|1.68μs|7.96ns|45.1ns|0.0105| 0|0|784&nbsp;B|
|#2399|`CallElasticsearchAsync`|net472|2.54μs|12.6ns|60.5ns|0.145| 0|0|931&nbsp;B|
|#2399|`CallElasticsearchAsync`|netcoreapp3.1|1.83μs|8.74ns|36.1ns|0.0123| 0|0|904&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`ExecuteAsync`|net472|3.23μs|16.1ns|68.5ns|0.164| 0|0|1.05&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`ExecuteAsync`|netcoreapp3.1|1.88μs|6.75ns|25.2ns|0.0139| 0|0|1.02&nbsp;KB|
|#2399|`ExecuteAsync`|net472|3.41μs|17.6ns|82.7ns|0.163| 0|0|1.05&nbsp;KB|
|#2399|`ExecuteAsync`|netcoreapp3.1|1.88μs|9.92ns|58.7ns|0.0143| 0|0|1.02&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #2399

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.HttpClientBenchmark.SendAsync&#8209;net472 | 1.151 | 6,227.91 | 7,167.27 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`SendAsync`|net472|6.25μs|32ns|160ns|0.353| 0|0|2.26&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`SendAsync`|netcoreapp3.1|4.67μs|17.2ns|64.3ns|0.03| 0|0|2.16&nbsp;KB|
|#2399|`SendAsync`|net472|7.18μs|28.6ns|111ns|0.352| 0|0|2.26&nbsp;KB|
|#2399|`SendAsync`|netcoreapp3.1|4.33μs|15ns|58ns|0.0299| 0|0|2.16&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`EnrichedLog`|net472|3.49μs|17.4ns|75.7ns|0.225| 0|0|1.44&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`EnrichedLog`|netcoreapp3.1|3.12μs|15.1ns|64ns|0.0204| 0|0|1.52&nbsp;KB|
|#2399|`EnrichedLog`|net472|3.47μs|18.5ns|96ns|0.225| 0|0|1.44&nbsp;KB|
|#2399|`EnrichedLog`|netcoreapp3.1|3.16μs|17.1ns|96.9ns|0.0206| 0|0|1.52&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`EnrichedLog`|net472|288μs|1.17μs|4.52μs|0.43| 0.143|0|4.31&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`EnrichedLog`|netcoreapp3.1|234μs|732ns|2.84μs|0| 0|0|4.19&nbsp;KB|
|#2399|`EnrichedLog`|net472|291μs|1.2μs|4.66μs|0.435| 0.145|0|4.31&nbsp;KB|
|#2399|`EnrichedLog`|netcoreapp3.1|232μs|744ns|2.88μs|0| 0|0|4.19&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`EnrichedLog`|net472|8.13μs|41.1ns|184ns|0.504| 0|0|3.21&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`EnrichedLog`|netcoreapp3.1|6.34μs|34.8ns|217ns|0.0484| 0|0|3.58&nbsp;KB|
|#2399|`EnrichedLog`|net472|7.57μs|32.9ns|127ns|0.504| 0|0|3.21&nbsp;KB|
|#2399|`EnrichedLog`|netcoreapp3.1|6.39μs|15.2ns|58.9ns|0.0503| 0|0|3.58&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`SendReceive`|net472|2.07μs|6.92ns|25.9ns|0.157| 0|0|1&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`SendReceive`|netcoreapp3.1|1.96μs|5.56ns|21.5ns|0.0134| 0|0|1&nbsp;KB|
|#2399|`SendReceive`|net472|2.29μs|12.4ns|68ns|0.157| 0|0|1&nbsp;KB|
|#2399|`SendReceive`|netcoreapp3.1|1.93μs|7.76ns|30.1ns|0.0134| 0|0|1&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`EnrichedLog`|net472|5.92μs|31ns|152ns|0.29| 0|0|1.85&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`EnrichedLog`|netcoreapp3.1|5.07μs|25.4ns|108ns|0.0199| 0|0|1.47&nbsp;KB|
|#2399|`EnrichedLog`|net472|5.96μs|27.8ns|115ns|0.289| 0|0|1.85&nbsp;KB|
|#2399|`EnrichedLog`|netcoreapp3.1|5.15μs|27.6ns|154ns|0.0202| 0|0|1.47&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`StartFinishSpan`|net472|912ns|4.13ns|23.4ns|0.0703| 0|0|449&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`StartFinishSpan`|netcoreapp3.1|975ns|4.71ns|18.3ns|0.00585| 0|0|448&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`StartFinishScope`|net472|1.1μs|5.58ns|25ns|0.0829| 0|0|530&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/0fbc6620ad774f009e81aac9f1b05abc698179aa)|`StartFinishScope`|netcoreapp3.1|1.02μs|3.86ns|14.9ns|0.00796| 0|0|568&nbsp;B|
|#2399|`StartFinishSpan`|net472|890ns|2.26ns|8.45ns|0.0705| 0|0|449&nbsp;B|
|#2399|`StartFinishSpan`|netcoreapp3.1|925ns|5.11ns|29.4ns|0.00597| 0|0|448&nbsp;B|
|#2399|`StartFinishScope`|net472|1.08μs|4.53ns|17.6ns|0.0829| 0|0|530&nbsp;B|
|#2399|`StartFinishScope`|netcoreapp3.1|1.12μs|6.42ns|46.8ns|0.0076| 0|0|568&nbsp;B|
</details>

**Comment by lucaspimentel** [comment]
Superseded by #2617

**Pull request closed without merging**